lib/lhm/connection.rb

```diff
@@ -1,13 +1,17 @@
 require 'delegate'
+require 'forwardable'
 require 'lhm/sql_retry'
 
 module Lhm
     # Lhm::Connection inherits from SingleDelegator. It will forward any unknown method calls to the ActiveRecord
     # connection.
   class Connection < SimpleDelegator
+    extend Forwardable
 
-    def initialize(connection:, default_log_prefix: nil, options: {})
-      @default_log_prefix = default_log_prefix
+    alias ar_connection __getobj__
+    def_delegators :@sql_retry, :reconnect_with_consistent_host, :reconnect_with_consistent_host=, :retry_config=
```

Comment:
A minor thing: as a person that's not very familiar with `SingleDelegator`, this looks very magical at first glance, especially since we should not doing something that complicated..?